src/main/java/com/orange/main/products/service/impl/ProductsServiceImpl.java

```diff
@@ -32,19 +39,42 @@ public Products getProductById(Long productId) {
     }
 
     @Override
-    public List<Products> getAllProducts(Pageable page, List<String> fields) {
+    public List<ProductDTO> getAllProducts(Pageable page, List<String> fields) {
+        List<ProductDTO> dto = new ArrayList<>();
         Page<Products> products = productsRepository.findAll(page);
         //fields filiter
         List<Products> rtn = products.getContent().stream().map(p -> handleUnspecifiedColumns(p, fields)).toList();
-        return rtn;
+
+        if(!CollectionUtils.isEmpty(rtn)){
+            for(Products product:rtn){
+                Price p = product.getPrices()
+                .stream()
+                .min(Comparator.comparing(Price::getPrice))
+                .orElseThrow(NoSuchElementException::new);
+                ProductDTO productDto = new ProductDTO(product, p.getSellerType().toString(), p.getPrice(), p.getDiscount(), null==product.getProductImgs()?null:product.getProductImgs().getPath());
+                dto.add(productDto);
+           }
+        }
+        return dto;
     }
 
     @Override
-    public List<Products> findByNameIn(Pageable page, List<String> name, List<String> fields) {
+    public List<ProductDTO> findByNameIn(Pageable page, List<String> name, List<String> fields) {
+        List<ProductDTO> dto = new ArrayList<>();
         Page<Products> products = productsRepository.findByNameIn(page, name);
         //fields filiter
         List<Products> rtn = products.getContent().stream().map(p -> handleUnspecifiedColumns(p, fields)).toList();
-        return rtn;
+         if(!CollectionUtils.isEmpty(rtn)){
+            for(Products product:rtn){
+                Price p = product.getPrices()
+                .stream()
+                .min(Comparator.comparing(Price::getPrice))
+                .orElseThrow(NoSuchElementException::new);
+                ProductDTO productDto = new ProductDTO(product, p.getSellerType().toString(), p.getPrice(), p.getDiscount(), null==product.getProductImgs()?null:product.getProductImgs().getPath());
+                dto.add(productDto);
+           }
+        }
+        return dto;
```

Comment:
Using early return instead.

```suggestion
        if (CollectionUtils.isEmpty(rtn)) {
            return dto;
        }
        
        for(Products product : rtn) {
                Price p = product.getPrices()
                .stream()
                .min(Comparator.comparing(Price::getPrice))
                .orElseThrow(NoSuchElementException::new);
                ProductDTO productDto = new ProductDTO(product, p.getSellerType().toString(), p.getPrice(), p.getDiscount(), null==product.getProductImgs()?null:product.getProductImgs().getPath());
                dto.add(productDto);
        }
        return dto;
```